.github/workflows/build_and_run_idf_examples.yml

```diff
@@ -0,0 +1,76 @@
+name: ESP-IDF USB Device examples
+
+on:
+  pull_request:
+    types: [opened, reopened, synchronize]
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        idf_ver: ["release-v5.0", "release-v5.1", "release-v5.2", "release-v5.3", "release-v5.4", "latest"]
+    runs-on: ubuntu-20.04
+    container: espressif/idf:${{ matrix.idf_ver }}
+    env:
+      IDF_PATH: /opt/esp/idf
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: 'true'
+      - name: Build
+        shell: bash
+        run: |
+          . ${IDF_PATH}/export.sh
+          pip install idf-component-manager==1.5.2 idf-build-apps==2.4.3 --upgrade
+          export EXAMPLES_PATH="${IDF_PATH}/examples/peripherals/usb/device"
+          export MANIFEST_FILE="${IDF_PATH}/examples/peripherals/.build-test-rules.yml"
+          python .github/ci/override_managed_component.py tinyusb . ${EXAMPLES_PATH}/tusb_*
+          idf-build-apps find --path ${EXAMPLES_PATH}/ --recursive --target esp32s2 --manifest-file ${MANIFEST_FILE}
+          idf-build-apps build --path ${EXAMPLES_PATH}/ --recursive --target esp32s2 --manifest-file ${MANIFEST_FILE}
+      - uses: actions/upload-artifact@v4
+        with:
+          name: usb_device_tusb_apps_bin_${{ matrix.idf_ver }}
+          path: |
+              /opt/esp/idf/examples/peripherals/usb/device/tusb_*/build/bootloader/bootloader.bin
+              /opt/esp/idf/examples/peripherals/usb/device/tusb_*/build/partition_table/partition-table.bin
+              /opt/esp/idf/examples/peripherals/usb/device/tusb_*/build/tusb_*.bin
+              /opt/esp/idf/examples/peripherals/usb/device/tusb_*/build/tusb_*.elf
+              /opt/esp/idf/examples/peripherals/usb/device/tusb_*/build/flasher_args.json
+              /opt/esp/idf/examples/peripherals/usb/device/tusb_*/build/config/sdkconfig.json
+          if-no-files-found: error
+  run-target:
+    name: Run
+    if: ${{ github.repository_owner == 'espressif' }}
+    needs: build
+    strategy:
+      fail-fast: false
+      matrix:
+        idf_ver: ["release-v5.3", "release-v5.4", "latest"]
```

Comment:
Just an FMI, which idf versions do we want to run on target?

Reply:
By the backport policy, usually we don't backport the examples. 
Only when it is required by fixing or adding features, so the pytest isn't working on 5.0 (but I didn't check the rest). 
I remember that I did backport some of the examples to 5.3 and 5.4 and I just started basically from 5.3, so the on 5.3 everything works.

Reply:
Just asking, if there is any reason behind having build matrix for IDF 5.0 to the IDF latest and having run matrix only for IDF 5.3 to the IDF latest. 
In esp-usb for example, we have both, the build and the run for all active IDF releases.

Reply:
While I decided to add the CI verification, I thought about:
1. Build for everything, that could be built with  the tinyusb component to be sure, that we won't break any build with the release (like build one example from esp-iot, because ATM the only one example could be built with local component usage)
2. Run at least as much pytest as we can (or we have ATM). To be sure, that I don't need to run the pytest locally everytime after release. 

Sure, it might be better (with more test_apps and better test case scenarios), but my general plan for this MR was to make automatically everything that might be run locally.